app/controllers/admin/switch_users_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Admin
+  class SwitchUsersController < ApplicationController
+    before_action :login_required
```

Comment:
Just thinking out loud, not making a recommendation...  I think I understand that you can get away with the RestrictAccessToAdminMode mechanism because your new controllers are all so small that each controller is more or less guaranteed to be entirely admin-required or not.  Okay, so then what about login_required?  How often are the new controllers divided where some methods are open access and others require login?  Or is it just more complicated when it comes to login, e.g. for login and verification controllers?

Reply:
`RestrictAccessToAdminMode` basically just changes two parameters of `login_required`. 
`login_required` still has to be a `before_action` on the controller, for the mechanism to work. 

But that might not be the nature of your question!

Reply:
Oh, good point.  There is probably some magic you can do with modules to have them "run a class method" in the requirer each time the module is required.  That is, have it add a login_required before filter in the requiring controller.  If it is important to anyone (hah!) it can be done just fine in a future PR. :)